std/http/file_server.ts

```diff
@@ -306,7 +321,19 @@ function html(strings: TemplateStringsArray, ...values: unknown[]): string {
 
 function main(): void {
   const CORSEnabled = serverArgs.cors ? true : false;
-  const addr = `0.0.0.0:${serverArgs.port ?? serverArgs.p ?? 4507}`;
+  const port = serverArgs.port ?? serverArgs.p ?? 4507;
+  const host = serverArgs.host ?? "0.0.0.0";
+  const addr = `${host}:${port}`;
+  const tlsOpts = {} as HTTPSOptions;
+  tlsOpts.certFile = serverArgs.cert ?? serverArgs.c ?? "";
+  tlsOpts.keyFile = serverArgs.key ?? serverArgs.k ?? "";
+
+  if (tlsOpts.keyFile || tlsOpts.certFile) {
+    if (tlsOpts.keyFile === "" || tlsOpts.certFile === "") {
+      console.log("--host, --key and --cert are required for TLS");
```

Comment:
Why is `--host` required for https? This is the bind address, it does not change how https functions.

Reply:
if you use certificates, (which I do) you'll want to specify the host that listens matching the certificate you have.

Reply:
The host you are passing in here is the IP or name of the the network interface you want to bind to and is not used for HTTPS. The server does not need to know what domain name it is serving a request for. Only the client uses the host name (which doesn't need to be sent to the server) to validate that the certificate is valid (by comparing it to the common name or subject alternative names in the certificate).

Reply:
Agreed - but if the service is only valid from a particular name, serving it under all interfaces will miserably fail on all the ones that don't match, instead of just simply not being there. Used to coding servers that behave this way, but I can certainly remove the host flag.

Reply:
To clarify I don't want to remove the --host flag. I just don't think it makes sense in this warning message because it is not required for https.

Reply:
Ah - very good - I am perfectly happy with that.